src/content/tutorials/github-twilio-notifications/index.md

```diff
@@ -0,0 +1,287 @@
+
+Finally, we will send a text message to you with the Github repo update info using Twilio. You'll need a Twilio account and number to send texts so follow [this guide to get setup](https://www.twilio.com/sms/api). FYI, Twilio has this [pretty awesome game](https://www.twilio.com/quest) where you can learn how to use their platform and get some free credits while you are at it.
+
+Then, we can create a helper function to send text messages for us by sending a post request to the Twilio API endpoint. [Refer to the Twilio reference here](https://www.twilio.com/docs/sms/api/message-resource#create-a-message-resource)
+
+We construct our headers and body in the format show n in the Twilio reference. Change the from value to your Twilio Phone number.
+
+```javascript
+---
+filename: index.js - sendText()
+---
+async function sendText(message){
+  let headers = new Headers();
+  headers.append("Authorization","Basic " + Buffer.from(accountSid + ":" + authToken).toString('base64'))
+  headers.append("Content-Type", "application/x-www-form-urlencoded")
+
+  const endpoint = "https://api.twilio.com/2010-04-01/Accounts/" + accountSid + "/Messages.json"
+
+  let encoded = new URLSearchParams()
+  encoded.append("To",recipient)
+  encoded.append("From", '+19388887573')
+  encoded.append("Body", message)
+
+  const request = {
+    body: encoded,
+    method: 'POST',
+    headers: headers
+  }
+
+  let result = await fetch(endpoint, request)
+  result = await result.json()
+
+  return new Response(JSON.stringify(result), request)
+}
+```
+In order to make this work, we need to set some secrets to hide your account SID and authToken from the source code. You can set secrets as such to get a prompt for the value of the secret name you input:
+
+```sh
+wrangler secret put accountSID
+```
+
+Use this syntax to set your Twilio accountSID, recipient(your number), Twilio authToken, and SECRET_TOKEN (for Github) to the respecive values.
+
+Finally, we modify our `githubWebhookHandler` to send a text at the end.
+
+```javascript
+async function githubWebhookHandler(request) {
+  if (request.method !== "POST") {
+    return simpleResponse(
+      200,
+      `Please send a POST request :)`
+    )
+  }
+  try {
+    if (!checkSignature(request)) {
+      return simpleResponse(403, "Incorrect Secret Code")
+    }
+
+    const formData = await request.json()
+    const repo_name = formData.repository.full_name
+    const action = await request.headers.get("X-GitHub-Event")
+    const sender_name = formData.sender.login
+    
+    return await sendText(`${sender_name} performed ${action} on ${repo_name}`)
+
+  } catch (e) {
+    return simpleResponse(
+      200,
+      `Error:  ${e} `
+    )
+  }
+}
+```
+
+Congrats, you finished this tutorial! Test it out and make a push to your repo and you should get a text message soon after. If you've never used Git before, here's a [quick guide](https://www.datacamp.com/community/tutorials/git-push-pull) to pushing to your repo. 
```

Comment:
idea that would be awesome but no pressure: a GIF of your phone receiving a text message from this worker :)

Reply:
also would recommend linking to the final repo again here!